pilot/pkg/model/push_context.go

```diff
@@ -2315,6 +2315,11 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) *MergedEnvoyFilterWrapper {
 func (ps *PushContext) getMatchedEnvoyFilters(proxy *Proxy, namespaces string) []*EnvoyFilterWrapper {
 	matchedEnvoyFilters := make([]*EnvoyFilterWrapper, 0)
 	for _, efw := range ps.envoyFiltersByNamespace[namespaces] {
+		if efw.GetTargetRefs() != nil {
```

Comment:
Does this mean `targetRefs` just doesn't work?

That's not obvious in the docs here

https://istio.io/latest/docs/reference/config/networking/envoy-filter/#EnvoyFilter